doc/design/logs.md

```diff
@@ -0,0 +1,35 @@
+# Logs in SQLFlow
+## Motivation
+In order to know well about the runtime status of the SQLFlow job, we need to count the number of `TRAIN/PREDICT/EXPLAIN/NORMAL` tasks over a period of time. For example, using ELK Stack for log query and analysis‎. Generally, such statistics components are implemented by logs.
+
+## logging library
+Several statistics components use `logtail` to collect logs. The `logtail` demands a file. If such information is written to `stdout`, we should redirect the `stdout` to a local file, the file size will increase until the SQLFlow server terminates, so we need a rolling file to store the log. Besides, a well-formatted log is readable.
+We use [Logrus](https://github.com/sirupsen/logrus) for layout, combining with [https://github.com/natefinch/lumberjack] for rolling the log file.
```

Comment:
Please note the reasoning of choosing Logrus over the standard log package. https://github.com/sql-machine-learning/sqlflow/issues/781#issuecomment-528211632

One reason I can think of is Logrus guarantees one-log-one-line, which the standard log package doesn't.

```go
package main

import "log"

func main() {
	log.SetFlags(log.Lshortfile | log.LstdFlags)
	log.Printf(`multiple
lines
logs`)
/* prints 2020/03/09 17:07:17 scratch.go:8: multiple
lines
logs
*/

}
```